src/dataExplorer/components/TagSelector.tsx

```diff
@@ -0,0 +1,172 @@
+import React, {FC, useContext, useEffect, useState} from 'react'
+import {concat, slice} from 'lodash'
+
+// Components
+import {Accordion} from '@influxdata/clockface'
+import SelectorTitle from 'src/dataExplorer/components/SelectorTitle'
+import WaitingText from 'src/shared/components/WaitingText'
+
+// Contexts
+import {
+  LOCAL_LIMIT,
+  NewDataExplorerContext,
+} from 'src/dataExplorer/context/newDataExplorer'
+
+// Types
+import {RemoteDataState} from 'src/types'
+
+interface Prop {
+  loading: RemoteDataState
+  tagKey: string
+  tagValues: string[]
+  onSelectTagKey: (key: string) => void
+  onSelectTagValue: (value: string) => void
+}
+
+const TagValues: FC<Prop> = ({
+  loading,
+  tagKey,
+  tagValues,
+  onSelectTagKey,
+  onSelectTagValue,
+}) => {
+  const [valuesToShow, setValuesToShow] = useState([])
+  const [loadMore, setLoadMore] = useState(false)
+  const [index, setIndex] = useState(LOCAL_LIMIT)
+
+  useEffect(() => {
+    if (tagValues.length === 0) {
+      // Reset
+      setValuesToShow([])
+      setLoadMore(false)
+      setIndex(LOCAL_LIMIT)
+    } else {
+      const newValuesToShow = slice(tagValues, 0, index)
+      const newLoadMore = tagValues.length > LOCAL_LIMIT
+      setValuesToShow(newValuesToShow)
+      setLoadMore(newLoadMore)
+    }
+  }, [tagValues])
+
+  let list: JSX.Element | JSX.Element[] = []
+
+  if (loading === RemoteDataState.Error) {
+    list = (
+      <div className="tag-selector-value--list-item">
+        Failed to load tag values
+      </div>
+    )
+  } else if (
+    loading === RemoteDataState.Loading ||
+    loading === RemoteDataState.NotStarted
+  ) {
+    list = <WaitingText text="Loading tag values" />
+  } else if (loading === RemoteDataState.Done && tagValues.length) {
+    list = valuesToShow.map(value => (
+      <div
+        className="tag-selector-value--list-item"
+        key={value}
+        onClick={() => onSelectTagValue(value)}
+      >
+        {value}
+      </div>
+    ))
+  }
+
+  const handleLoadMore = () => {
+    const newIndex = index + LOCAL_LIMIT
+    if (loadMore) {
+      // Add more key values to show
+      const newValuesToShow = concat(
+        valuesToShow,
+        slice(tagValues, index, newIndex)
+      )
+      setValuesToShow(newValuesToShow)
+    }
+    const newLoadMore = newIndex < tagValues.length
+    setLoadMore(newLoadMore)
+    setIndex(newIndex)
+  }
+
+  const loadMoreButton = loadMore && (
+    <div className="load-more-button" onClick={handleLoadMore}>
+      + Load more
+    </div>
+  )
+
+  return (
+    <Accordion className="tag-selector-value">
+      <Accordion.AccordionHeader className="tag-selector-value--header">
+        <div onClick={() => onSelectTagKey(tagKey)}>
+          <SelectorTitle title={tagKey} />
+        </div>
+      </Accordion.AccordionHeader>
+      <div className="container-side-bar">
```

Comment:
based on the video it looks like the list text is a bit squished, any chance we can add a larger margin-bottom to this?